input_writer.go

```diff
@@ -0,0 +1,30 @@
+package logrus
+
+import (
+	"bufio"
+	"io"
+	"runtime"
+)
+
+func (logger *Logger) InputWriter() (*io.PipeWriter) {
+	inputReader, inputWriter := io.Pipe()
+
+	go logger.inputWriterScanner(inputReader)
+	runtime.SetFinalizer(inputWriter, inputWriterFinalizer)
```

Comment:
But if someone does something such as:

```
log.SetOutput(logger.InputWriter())
...
log.SetOutput(somethingElse)
```

We'll leak a goroutine since the io.PipeWriter never gets closed (I tested this scenario). If the user wants to explicitly close the io.PipeWriter we give them, they can, but I didn't see any harm in closing it ourself if the garbage collector comes around and removes the InputWriter.